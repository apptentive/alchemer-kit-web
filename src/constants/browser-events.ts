/**
 * Constants
 */
export default {
  APPTENTIVE_APP_STORE_RATING_BLOCKED: 'apptentive:app-store-rating:blocked',
  APPTENTIVE_APP_STORE_RATING_ERROR: 'apptentive:app-store-rating:error',
  APPTENTIVE_APP_STORE_RATING_OPEN_APP_STORE_URL: 'apptentive:app-store-rating:open-app-store-url',
  APPTENTIVE_ERROR: 'apptentive:error',
  APPTENTIVE_INIT: 'apptentive:init',
  APPTENTIVE_READY: 'apptentive:ready',
  APPTENTIVE_LOVE_DIALOG_CANCEL: 'apptentive:love-dialog:cancel',
  APPTENTIVE_LOVE_DIALOG_CLOSE: 'apptentive:love-dialog:close',
  APPTENTIVE_LOVE_DIALOG_DISMISS: 'apptentive:love-dialog:dismiss',
  APPTENTIVE_LOVE_DIALOG_ERROR: 'apptentive:love-dialog:error',
  APPTENTIVE_LOVE_DIALOG_LAUNCH: 'apptentive:love-dialog:launch',
  APPTENTIVE_LOVE_DIALOG_NO: 'apptentive:love-dialog:no',
  APPTENTIVE_LOVE_DIALOG_YES: 'apptentive:love-dialog:yes',
  APPTENTIVE_MESSAGE_CENTER_CLOSE: 'apptentive:message-center:close',
  APPTENTIVE_MESSAGE_CENTER_DISMISS: 'apptentive:message-center:dismiss',
  APPTENTIVE_MESSAGE_CENTER_ERROR: 'apptentive:message-center:error',
  APPTENTIVE_MESSAGE_CENTER_LAUNCH: 'apptentive:message-center:launch',
  APPTENTIVE_MESSAGE_CENTER_SEND: 'apptentive:message-center:send',
  APPTENTIVE_MESSAGES_CREATED: 'apptentive:messages:created',
  APPTENTIVE_MESSAGES_FETCHED: 'apptentive:messages:fetched',
  APPTENTIVE_NAVIGATE_TO_LINK_BLOCKED: 'apptentive:navigate-to-link:blocked',
  APPTENTIVE_NAVIGATE_TO_LINK_ERROR: 'apptentive:navigate-to-link:error',
  APPTENTIVE_NAVIGATE_TO_LINK_NAVIGATE: 'apptentive:navigate-to-link:navigate',
  APPTENTIVE_NAVIGATE_TO_LINK_NAVIGATED: 'apptentive:navigate-to-link:navigated',
  APPTENTIVE_NOTE_CANCEL: 'apptentive:note:cancel',
  APPTENTIVE_NOTE_CLICK: 'apptentive:note:click',
  APPTENTIVE_NOTE_CLOSE: 'apptentive:note:close',
  APPTENTIVE_NOTE_DISMISS: 'apptentive:note:dismiss',
  APPTENTIVE_NOTE_ERROR: 'apptentive:note:error',
  APPTENTIVE_NOTE_LAUNCH: 'apptentive:note:launch',
  APPTENTIVE_OVERLAY_CLICK: 'apptentive:overlay:click',
  APPTENTIVE_RENDER_INTERACTION: 'apptentive:render:interaction',
  APPTENTIVE_SURVEY_DISMISS: 'apptentive:survey:dismiss',
  APPTENTIVE_SURVEY_ERROR: 'apptentive:survey:error',
  APPTENTIVE_SURVEY_LAUNCH: 'apptentive:survey:launch',
  APPTENTIVE_SURVEY_SUBMIT: 'apptentive:survey:submit',
  APPTENTIVE_SURVEY_NEXT_QUESTION_SET: 'apptentive:survey:next_question_set',
};
