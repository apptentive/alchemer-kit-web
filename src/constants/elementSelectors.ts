export enum surveySelectors {
  title = '.apptentive-appbar__title',
  answerChoice = 'answer-choice',
  answerChoices = 'answer-choices',
  errorMessage = 'apptentive-errors',
  invalidReason = 'invalid-reason',
  thankYou = 'apptentive-survey-thank-you',
  questionsForm = 'apptentive-survey-questions',
  questionsWrapper = 'apptentive-survey__question-wrapper',
  actionsWrapper = 'apptentive-survey__actions-wrapper',
}

export enum messageCenterSelectors {
  content = '.apptentive-message-center__content',
  feedback = '.apptentive-message-center-feedback',
  thankYou = 'apptentive-message-center-thank-you',
  profile = '.apptentive-message-center-profile',
  profileNameInput = '.apptentive-message-center-profile__name .apptentive-textinput',
  profileEmailInput = '.apptentive-message-center-profile__email .apptentive-textinput',
  submitButton = '.apptentive-message-center__actions .apptentive-button--primary',
}

export enum interactionSelectors {
  confirmBlock = 'apptentive-interaction__confirm-wrapper',
  blocked = 'apptentive-interaction__blocked',
}
