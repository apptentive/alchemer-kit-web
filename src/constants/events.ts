/**
 * Constants
 */
export default {
  APPTENTIVE_APP_STORE_RATING_LAUNCH: 'com.apptentive#AppStoreRating#launch',
  APPTENTIVE_APP_STORE_RATING_OPEN_APP_STORE_URL: 'com.apptentive#AppStoreRating#open_app_store_url',
  APPTENTIVE_APP_STORE_RATING_UNABLE_TO_RATE: 'com.apptentive#AppStoreRating#unable_to_rate',
  APPTENTIVE_INIT: 'com.apptentive#app#init',
  APPTENTIVE_LAUNCH: 'com.apptentive#app#launch',
  APPTENTIVE_LOVE_DIALOG_CANCEL: 'com.apptentive#EnjoymentDialog#cancel',
  APPTENTIVE_LOVE_DIALOG_DISMISS: 'com.apptentive#EnjoymentDialog#dismiss',
  APPTENTIVE_LOVE_DIALOG_LAUNCH: 'com.apptentive#EnjoymentDialog#launch',
  APPTENTIVE_LOVE_DIALOG_NO: 'com.apptentive#EnjoymentDialog#no',
  APPTENTIVE_LOVE_DIALOG_YES: 'com.apptentive#EnjoymentDialog#yes',
  APPTENTIVE_MESSAGE_CENTER_CLOSE: 'com.apptentive#MessageCenter#close',
  APPTENTIVE_MESSAGE_CENTER_LAUNCH: 'com.apptentive#MessageCenter#launch',
  APPTENTIVE_MESSAGE_CENTER_SEND: 'com.apptentive#MessageCenter#send',
  APPTENTIVE_NAVIGATE_TO_LINK_NAVIGATE: 'com.apptentive#NavigateToLink#navigate',
  APPTENTIVE_NOTE_CANCEL: 'com.apptentive#TextModal#cancel',
  APPTENTIVE_NOTE_CLICK_INTERACTION: 'com.apptentive#TextModal#interaction',
  APPTENTIVE_NOTE_DISMISS: 'com.apptentive#TextModal#dismiss',
  APPTENTIVE_NOTE_LAUNCH: 'com.apptentive#TextModal#launch',
  APPTENTIVE_SURVEY_CANCEL: 'com.apptentive#Survey#cancel',
  APPTENTIVE_SURVEY_CANCEL_PARTIALLY: 'com.apptentive#Survey#cancel_partial',
  APPTENTIVE_SURVEY_CLOSE: 'com.apptentive#Survey#close',
  APPTENTIVE_SURVEY_LAUNCH: 'com.apptentive#Survey#launch',
  APPTENTIVE_SURVEY_SUBMIT: 'com.apptentive#Survey#submit',
  APPTENTIVE_SURVEY_NEXT_QUESTION_SET: 'com.apptentive#Survey#next_question_set',
  APPTENTIVE_SURVEY_QUESTION_SET_SHOWN: 'com.apptentive#Survey#question_set_shown',
};
