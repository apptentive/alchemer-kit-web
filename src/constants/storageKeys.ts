export enum sessionStorageKeys {
  manifestCache = 'apptentiveCachedManifest',
  manifestExpiration = 'apptentiveCacheExpiration',
  manifestLocale = 'apptentiveLocaleOverride',
}

export enum localStorageKeys {
  appId = 'app_id',
  captureDisabled = 'captureDisabled',
  codePoint = 'code_point',
  conversation = 'conversation',
  device = 'device',
  interactionCounts = 'interaction_counts',
  person = 'person',
  random = 'random',
}
