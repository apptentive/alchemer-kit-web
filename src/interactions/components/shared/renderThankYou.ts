/**
 * Renders a given surveys Success Message / 'Thank You' Message.
 *
 * @param {Function} onDismiss - The method to call when the dismiss button is clicked
 * @param {string} customElementName - The custom element to create for the content
 * @param {string} content - The text content to display on the thankyou screen
 * @param {string} buttonText - The text within the dismiss button
 * @param {boolean} isTall - Whether or not to increase the spacing between the bottom and button
 * @returns {HTMLElement} - The 'Thank You' DOM Node
 */
const renderThankYou = (
  onDismiss: VoidFunction,
  customElementName: string,
  content = 'Thank you!',
  buttonText = 'Close',
  isTall = false
) => {
  const thankYouContentElement = document.createElement('p');
  thankYouContentElement.classList.add('apptentive-message__content');
  thankYouContentElement.textContent = content;

  const thankYouWrapperElement = document.createElement('div');
  thankYouWrapperElement.classList.add('apptentive-message__content-wrapper');
  thankYouWrapperElement.append(thankYouContentElement);

  const dismissButtonText = buttonText;
  const thankYouDismissButton = document.createElement('button');
  thankYouDismissButton.classList.add('apptentive-button--primary');
  thankYouDismissButton.textContent = dismissButtonText;
  thankYouDismissButton.title = dismissButtonText;
  thankYouDismissButton.addEventListener('click', onDismiss);

  const thankYouActionsElement = document.createElement('div');
  thankYouActionsElement.classList.add('apptentive-message__actions');

  if (isTall) {
    thankYouActionsElement.classList.add('apptentive-message__actions--tall');
  }

  thankYouActionsElement.append(thankYouDismissButton);

  const thankYouElement = document.createElement(customElementName);
  thankYouElement.style.display = 'none';
  thankYouElement.classList.add('apptentive-message');
  thankYouElement.append(thankYouWrapperElement);
  thankYouElement.append(thankYouActionsElement);

  return thankYouElement;
};

export default renderThankYou;
