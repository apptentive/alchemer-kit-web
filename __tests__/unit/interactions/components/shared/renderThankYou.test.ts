import renderThankYou from '../../../../../src/interactions/components/shared/renderThankYou';
import { surveySelectors } from '../../../../../src/constants/elementSelectors';

const _contentText = 'Thank you for your feedback!';
const _buttonText = 'Close Survey';
const _onDismiss = jest.fn();

describe('renderThankYou', () => {
  afterEach(() => {
    document.body.innerHTML = '';
    _onDismiss.mockClear();
  });

  test('properly renders tall thank you screen', () => {
    const output = renderThankYou(_onDismiss, surveySelectors.thankYou, _contentText, _buttonText, true);

    expect(output).toMatchSnapshot();
  });

  test('properly renders short thank you screen', () => {
    const output = renderThankYou(_onDismiss, surveySelectors.thankYou, _contentText, _buttonText, false);

    expect(output).toMatchSnapshot();
  });

  test('properly renders thank you with defaults', () => {
    const output = renderThankYou(_onDismiss, surveySelectors.thankYou);

    expect(output).toMatchSnapshot();
  });

  test('properly attaches dismiss function', () => {
    document.body.append(renderThankYou(_onDismiss, surveySelectors.thankYou, _contentText, _buttonText, false));

    const dismissButton = document.querySelector<HTMLButtonElement>('.apptentive-button--primary')!;
    dismissButton.click();

    expect(_onDismiss).toHaveBeenCalledTimes(1);
  });
});
