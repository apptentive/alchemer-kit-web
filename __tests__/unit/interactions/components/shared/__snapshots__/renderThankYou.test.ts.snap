// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderThankYou properly renders short thank you screen 1`] = `
<apptentive-survey-thank-you
  class="apptentive-message"
  style="display: none;"
>
  <div
    class="apptentive-message__content-wrapper"
  >
    <p
      class="apptentive-message__content"
    >
      Thank you for your feedback!
    </p>
  </div>
  <div
    class="apptentive-message__actions"
  >
    <button
      class="apptentive-button--primary"
      title="Close Survey"
    >
      Close Survey
    </button>
  </div>
</apptentive-survey-thank-you>
`;

exports[`renderThankYou properly renders tall thank you screen 1`] = `
<apptentive-survey-thank-you
  class="apptentive-message"
  style="display: none;"
>
  <div
    class="apptentive-message__content-wrapper"
  >
    <p
      class="apptentive-message__content"
    >
      Thank you for your feedback!
    </p>
  </div>
  <div
    class="apptentive-message__actions apptentive-message__actions--tall"
  >
    <button
      class="apptentive-button--primary"
      title="Close Survey"
    >
      Close Survey
    </button>
  </div>
</apptentive-survey-thank-you>
`;

exports[`renderThankYou properly renders thank you with defaults 1`] = `
<apptentive-survey-thank-you
  class="apptentive-message"
  style="display: none;"
>
  <div
    class="apptentive-message__content-wrapper"
  >
    <p
      class="apptentive-message__content"
    >
      Thank you!
    </p>
  </div>
  <div
    class="apptentive-message__actions"
  >
    <button
      class="apptentive-button--primary"
      title="Close"
    >
      Close
    </button>
  </div>
</apptentive-survey-thank-you>
`;
